src/backend/access/common/reloptions_gp.c

```diff
@@ -1358,6 +1358,19 @@ reloptions_has_opt(List *opts, const char *name)
 	return false;
 }
 
+DefElem *
+reloptions_get_opt(List *opts, const char *name)
+{
+	ListCell *lc;
+	foreach(lc, opts)
+	{
+		DefElem *de = lfirst(lc);
+		if (pg_strcasecmp(de->defname, name) == 0)
+			return de;
+	}
+	return NULL;
+}
+
```

Comment:
can this function be merged with `reloptions_has_opt`? they do the same thing and the returned pointer can simply be used to check for the existence of a option.

Reply:
Yes, we can replace the reloptions_has_opt function with reloptions_get_opt, but this will be a API change and does not directly relate to the current patch.